miniapp/setup_kernels.cxx

```diff
@@ -383,6 +383,98 @@ namespace {
    } // end setup_data_cuda
 
 #endif
+
+#if defined(RAJA_ENABLE_HIP)
+
+   void setup_data_hip(const int nqpts, const int nstatev,
+                        const double dt, const double* vel_grad_array,
+                        const double* stress_array, const double* state_vars_array,
+                        double* stress_svec_p_array, double* d_svec_p_array,
+                        double* w_vec_array, double* ddsdde_array,
+                        double* vol_ratio_array, double* eng_int_array,
+                        double* temp_array){
+      // vgrad is kinda a pain to deal with as a raw 1d array, so we're
+      // going to just use a RAJA view here. The data is taken to be in col. major format.
+      // It might be nice to eventually create a type alias for the below or
+      // maybe something like it.
+
+      const int ind_int_eng = nstatev - ecmech::ne;
+      const int ind_vols = ind_int_eng - 1;
+
+      const int DIM = 3;
+      std::array<RAJA::idx_t, DIM> perm { { 2, 1, 0 } };
+      RAJA::Layout<DIM> layout = RAJA::make_permuted_layout({ { ecmech::ndim, ecmech::ndim, nqpts } }, perm);
+      RAJA::View<const double, RAJA::Layout<DIM, RAJA::Index_type, 0> > vgrad_view(vel_grad_array, layout);
+
+      // All of the below we could setup in one big RAJA loop/kernel
+      RAJA::RangeSegment default_range(0, nqpts);
+
+      RAJA::forall<RAJA::hip_exec<384> >(default_range, [ = ] RAJA_DEVICE(int i_qpts) {
```

Comment:
his is a lot of code duplication for what appears to be a on-line difference in the RAJA::forall invocation.

Is there some not-overly-painful way to template this to avoid the code duplication?